### PR TITLE
Handle GT_FTN_ADDR in gtClone

### LINKED ISSUE #87888: Test failure: JIT/Regression/JitBlue/Runtime_72133/Runtime_72133/Runtime_72133.dll
windows arm64/x64 failures in random jitstress:

https://dev.azure.com/dnceng-public/public/_build/results?buildId=315671&view=ms.vss-test-web.build-test-results-tab&runId=6470456&resultId=120090&paneView=debug

win-arm64:
```
set DOTNET_TieredCompilation=0
set DOTNET_JitStress=17c
...
19:10:48.963 Passed test: JIT/Regression/JitBlue/Runtime_72133/Runtime_72133/Runtime_72133.dll
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Program.<<Main>$>g__TestExecutor264|0_265(System.IO.StreamWriter, System.IO.StreamWriter, <>c__DisplayClass0_0 ByRef)
   at Program.<Main>$(System.String[])
App Exit Code: -1073741819
Expected: 100
Actual: -1073741819
END EXECUTION - FAILED
FAILED
```

win-x64:
```
set DOTNET_TieredCompilation=0
set DOTNET_JitStress=97
```

@dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by MichalPetryka**
Fixes #87888

cc @jakobbotsch

### CODE REVIEW

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -8652,6 +8652,15 @@ GenTree* Compiler::gtClone(GenTree* tree, bool complexOK)
             break;
         }
 
+        case GT_FTN_ADDR:
+        {
+            copy = new (this, tree->OperGet()) GenTreeFptrVal(tree->TypeGet(), tree->AsFptrVal()->gtFptrMethod);
```
Comment (BruceForstall) [inline src/coreclr/jit/gentree.cpp:8657]: Looks like it also needs to unconditionally copy `gtFptrDelegateTarget`:
```
copy->AsFptrVal()->gtFptrDelegateTarget = tree->AsFptrVal()->gtFptrDelegateTarget;
```

Comment (MichalPetryka) [reply src/coreclr/jit/gentree.cpp:8657]: `gtCloneExpr` doesn't copy that, should that be added there too?

Comment (BruceForstall) [reply src/coreclr/jit/gentree.cpp:8657]: Yes.

Comment (MichalPetryka) [reply src/coreclr/jit/gentree.cpp:8657]: Done

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**